BUILD

```diff
@@ -86,6 +88,19 @@ genrule(
     ]),
 )
 
+genrule(
+    name = "generate_dist_lockfile",
+    srcs = [
+        "MODULE.bazel",
+        "//third_party/googleapis:MODULE.bazel",
+        "//third_party/remoteapis:MODULE.bazel",
+    ],
+    outs = ["MODULE.bazel.lock.dist"],
+    cmd = "touch BUILD && trap 'rm -rf tmp_bazel_root' EXIT && $(location //src:bazel) --output_user_root=$$PWD/tmp_bazel_root query --check_direct_dependencies=error --lockfile_mode=update :all && mv MODULE.bazel.lock $@",
+    tags = ["allow-network"],
```

Comment:
Oh, also run `bazel shutdown` at the end, otherwise on Windows:
```
rm: cannot remove 'tmp_bazel_root/install/0f8e75f56b18f23e1f984c1baaf2d330/A-server.jar': Device or resource busy
rm: cannot remove 'tmp_bazel_root/install/0f8e75f56b18f23e1f984c1baaf2d330/embedded_tools/jdk/lib/modules': Device or resource busy
rm: cannot remove 'tmp_bazel_root/owe4iaif/java.log.bk-windows-mcnm.b.log.java.20240212-200207.3708': Device or resource busy
rm: cannot remove 'tmp_bazel_root/owe4iaif/server/jvm.out': Device or resource busy
```

Reply:
done (switched to `--batch` so we wouldn't need to shutdown after)